implementation/src/main/java/io/smallrye/config/SmallRyeConfigBuilder.java

```diff
@@ -124,6 +127,11 @@ public SmallRyeConfigBuilder withSources(Collection<ConfigSource> configSources)
         return this;
     }
 
+    public SmallRyeConfigBuilder withInterceptors(ConfigSourceInterceptor... interceptors) {
```

Comment:
I think we'll also need the service loader based `ConfigSourceInterceptorProvider` that was discussed earlier, though that can come in a subsequent iteration.

Reply:
Yes, I didn't bother about it, since it is just boilerplate and would take the focus out of the API.